components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -1118,6 +1124,35 @@ private class MonitoredItemWrapper {
                             eventFilter.AddSelectClause(ObjectTypeIds.ConditionType, "Retain");
                         }
                     }
+
+                    var sb = new StringBuilder();
+
+                    // let's loop thru the select clause and setup the field names
+                    foreach (var selectClause in eventFilter.SelectClauses) {
+                        sb.Clear();
+                        for (var i = 0; i < selectClause.BrowsePath.Count; i++) {
```

Comment:
Sanity check... can BrowsePath be null? Or is it maybe always set?

Reply:
Fixed